pkg/executor/executor.go

```diff
@@ -125,31 +127,48 @@ func (s *pythonExecutor) SaveModel(cl *ir.TrainStmt) error {
 	return m.Save(modelURI, s.Session)
 }
 
-func (s *pythonExecutor) runCommand(program string, logStderr bool) error {
+func (s *pythonExecutor) runProgram(program string, logStderr bool) error {
+	cmd := sqlflowCmd(s.Cwd, s.Db.DriverName)
+	cmd.Stdin = bytes.NewBufferString(program)
+
+	errorLog, e := s.runCommand(cmd, nil, logStderr)
+	if e != nil {
+		// return the diagnostic message
+		sub := rePyDiagnosis.FindStringSubmatch(errorLog)
+		if len(sub) == 2 {
+			return fmt.Errorf("%s", sub[1])
+		}
+		// if no diagnostic message, return the full stack trace
+		return fmt.Errorf("failed: %v\n%sGenerated Code:%[2]s\n%s\n%[2]sOutput%[2]s\n%[4]v", e, "==========", program, errorLog)
+	}
+	return nil
+}
+
+func (s *pythonExecutor) runCommand(cmd *exec.Cmd, context map[string]string, logStderr bool) (string, error) {
 	cw := &logChanWriter{wr: s.Writer}
 	defer cw.Close()
-	cmd := sqlflowCmd(s.Cwd, s.Db.DriverName)
+
+	for k, v := range context {
```

Comment:
How about just set `cmd.Env` instead of setting the environment variables of parent process?

Reply:
I think it would be fine. Setting the environment variable in the container is what we want.